src/linux/x86_64/kvm_cpu.rs

```diff
@@ -305,8 +306,14 @@ impl KvmCpu {
 		&mut self.vcpu
 	}
 
-	fn init(&mut self, entry_point: u64, stack_address: u64, cpu_id: u32) -> HypervisorResult<()> {
-		self.setup_long_mode(entry_point, stack_address, cpu_id)?;
+	fn init(
+		&mut self,
+		entry_point: u64,
+		stack_address: u64,
+		guest_address: u64,
```

Comment:
What is `guest_address`? I think we need some documentation here.

Reply:
I could not clarify this in time, but `guest_address` is necessary for setting the registers to their correct values.

This was **not** done for every architecture and operating system - it was only done on x86_64 Linux and, using guesswork, on x86_64 macOS - however, I had to put it in there everywhere nevertheless, even if it has no effect on some architectures.